crates/toml_edit/src/table.rs

```diff
@@ -397,6 +397,24 @@ impl Table {
     pub fn remove_entry(&mut self, key: &str) -> Option<(Key, Item)> {
         self.items.shift_remove(key).map(|kv| (kv.key, kv.value))
     }
+
+    /// Retains only the elements specified by the `keep` predicate.
+    ///
+    /// In other words, remove all pairs `(key, value)` for which
+    /// `keep(&key, &mut value)` returns `false`.
+    ///
+    /// The elements are visited in iteration order.
+    pub fn retain<F>(&mut self, mut keep: F)
+    where
+        F: FnMut(&str, &mut Value) -> bool,
+    {
+        self.items.retain(|key, item| {
+            item.value
+                .as_value_mut()
+                .map(|value| keep(key, value))
+                .unwrap_or(false)
+        });
```

Comment:
This one should be dealing with `Item`, not `Value`

Reply:
Applied
```diff
-    /// In other words, remove all pairs `(key, value)` for which
-    /// `keep(&key, &mut value)` returns `false`.
+    /// In other words, remove all pairs `(key, item)` for which
+    /// `keep(&key, &mut item)` returns `false`.
     ///
     /// The elements are visited in iteration order.
     pub fn retain<F>(&mut self, mut keep: F)
     where
-        F: FnMut(&str, &mut Value) -> bool,
+        F: FnMut(&str, &mut Item) -> bool,
     {
-        self.items.retain(|key, item| {
-            item.value
-                .as_value_mut()
-                .map(|value| keep(key, value))
-                .unwrap_or(false)
-        });
+        self.items
+            .retain(|key, key_value| keep(key, &mut key_value.value));
```

This got a bit weird `.retain(|key, key_value|  ... key_value.value)`, because this Table maps `key -> key_value`.